app/src/main/java/com/github/gogiffy/activities/SearchActivity.java

```diff
@@ -50,15 +50,32 @@ public void beforeTextChanged(CharSequence s, int start, int count, int after) {
 
     @Override
     public void onTextChanged(CharSequence s, int start, int before, int count) {
-        if (timer != null) {
-            timer.cancel();
+        String      invalid_variable_stye_here_its_too_long_too2 =    "" +
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "invalid_variable_stye_here_its_too_long_too2". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this unused "invalid_variable_stye_here_its_too_long_too2" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS1481)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS00117)